src/structure.js

```diff
@@ -68,6 +77,57 @@ Structure.prototype.cursor = function (path) {
   return Cursor.from(self.current, path, changeListener);
 };
 
+Structure.prototype.reference = function (path) {
+  var self = this, pathId = pathString(path);
+  var listenerNs = self._pathListeners[pathId];
+
+  var cursor = this.cursor(path);
+  var changeListener = function (newRoot, oldRoot, changedPath) { cursor = self.cursor(path); };
+
+  var referenceListeners = [changeListener];
+  this._pathListeners[path] = !listenerNs ? [changeListener] : listenerNs.concat(changeListener);
+
+  return {
+    onChange: function (newFn) {
+      if (typeof newFn !== 'function') return;
+      var fnIndex, listenerNs = self._pathListeners[pathId];
+
+      self._pathListeners[pathId] = !listenerNs ? [newFn] : listenerNs.concat(newFn);
+      fnIndex = self._pathListeners[pathId].indexOf(newFn);
+
+      referenceListeners = (referenceListeners || []).concat(newFn);
```

Comment:
But `referenceListeners` is never falsey... it's either an empty array, or a populated array. 

Anyways, my point, which I forgot to mention, was that `changeListener` gets removed after `unsubscribeAll()`, and doesn't get added back to both `self._pathListeners` and `referenceListeners`.


May get added to `self._pathListeners` on next `reference()` call.